proxy/http/HttpTransact.cc

```diff
@@ -2070,6 +2070,14 @@ HttpTransact::OSDNSLookup(State *s)
       } else if (s->cache_lookup_result == CACHE_LOOKUP_MISS || s->cache_info.action == CACHE_DO_NO_ACTION) {
         TRANSACT_RETURN(SM_ACTION_API_OS_DNS, HandleCacheOpenReadMiss);
         // DNS lookup is done if the lookup failed and need to call Handle Cache Open Read Miss
+      } else if (s->cache_info.action == CACHE_PREPARE_TO_WRITE && s->http_config_param->cache_post_method == 1 &&
```

Comment:
Why isn't this a logical or condition on the previous clause?

Reply:
Logically it can be combined. But note that this is actually the third condition that returns `TRANSACT_RETURN(SM_ACTION_API_OS_DNS, HandleCacheOpenReadMiss);`. Since this block of conditions is deriving from state how it got here, and therefore what should be done from here, each else clause corresponds to a different historical path. As such, most of the blocks have an explanatory comment, which I've now added for my new condition. I prefer this over combining these three conditions into one big conditional statement because it makes the various state transitions more comprehensible.